src/Aspire.Hosting.SqlServer/SqlServerBuilderExtensions.cs

```diff
@@ -28,6 +29,33 @@ public static IResourceBuilder<SqlServerServerResource> AddSqlServer(this IDistr
         var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password", minLower: 1, minUpper: 1, minNumeric: 1);
 
         var sqlServer = new SqlServerServerResource(name, passwordParameter);
+
+        string? connectionString = null;
+
+        builder.Eventing.Subscribe<ConnectionStringAvailableEvent>(sqlServer, async (@event, ct) =>
+        {
+            connectionString = await sqlServer.GetConnectionStringAsync(ct).ConfigureAwait(false);
```

Comment:
Should this fail if `connectionString` is null, and not publish `ConnectionStringAvailableEvent`?

Reply:
This is unlikely to occur but I think in the interests of being defensive its not a bad idea to do this.